sync3/caches/user.go

```diff
@@ -138,9 +138,9 @@ func NewInviteData(userID, roomID string, inviteState []json.RawMessage) *Invite
 		}
 	}
 	if id.InviteEvent == nil {
-		logger.Error().Str("invitee", userID).Str("room", roomID).Int("num_invite_state", len(inviteState)).Msg(
-			"cannot make invite, missing invite event for user",
-		)
+		const errMsg = "cannot make invite, missing invite event for user"
+		logger.Error().Str("invitee", userID).Str("room", roomID).Int("num_invite_state", len(inviteState)).Msg(errMsg)
+		internal.GetSentryHubFromContextOrDefault(ctx).CaptureException(fmt.Errorf(errMsg))
```

Comment:
Can we track the `num_invite_state` etc with this?

Reply:
Looks like a "context" on a "scope" is the way to do this:

E.g. with the test snippet

```go
	hub := internal.GetSentryHubFromContextOrDefault(ctx)
	hub.WithScope(func(scope *sentry.Scope) {
		scope.SetContext("sliding-sync", map[string]interface{}{
			"invitee":          "@dmr:test",
			"room":             "!dmr:test",
			"num_invite_state": 12345,
		})
		hub.CaptureException(fmt.Errorf("DMR TEST DMR TEST"))
	})
```

I get

![image](https://user-images.githubusercontent.com/8614563/231560166-7b0ce9ad-1134-4f21-95c7-787990110bda.png)

Reply:
(You can also write key-value string pairs which get indexed and are searchable but I didn't think that would be useful here.)